Type check and lint failing in default project

The project doesn't actually build out of the box, because the type check and lint tasks are failing.

I know nothing about react-query. Will plan on diving into that apparently complicated topic as quickly as I can, but in the mean time, the last type-checking errors that I'm struggling with are in `src/utils/api/use-team-role.ts`. If you have any quick tips on how to fix these errors, I would really appreciate it:
```
src/utils/api/use-team-role.ts:24:5 - error TS2769: No overload matches this call.
  Overload 1 of 9, '(queryKey: QueryKey, queryFn: QueryFunction<UseTeamRoleResponse, QueryKey>, options?: Omit<UseQueryOptions<UseTeamRoleResponse, Error, UseTeamRoleResponse, QueryKey>, "initialData" | ... 1 more ... | "queryKey"> & { ...; }): UseQueryResult<...>', gave the following error.
    Argument of type '() => Promise<Json>' is not assignable to parameter of type 'QueryFunction<UseTeamRoleResponse, QueryKey>'.
      Type 'Promise<Json>' is not assignable to type 'UseTeamRoleResponse | Promise<UseTeamRoleResponse>'.
        Type 'Promise<Json>' is not assignable to type 'Promise<UseTeamRoleResponse>'.
          Type 'Json' is not assignable to type 'UseTeamRoleResponse'.
            Type 'string' is not assignable to type 'UseTeamRoleResponse'.
  Overload 2 of 9, '(queryKey: QueryKey, queryFn: QueryFunction<UseTeamRoleResponse, QueryKey>, options?: Omit<UseQueryOptions<UseTeamRoleResponse, Error, UseTeamRoleResponse, QueryKey>, "initialData" | ... 1 more ... | "queryKey"> & { ...; }): DefinedUseQueryResult<...>', gave the following error.
    Argument of type '() => Promise<Json>' is not assignable to parameter of type 'QueryFunction<UseTeamRoleResponse, QueryKey>'.
  Overload 3 of 9, '(queryKey: QueryKey, queryFn: QueryFunction<UseTeamRoleResponse, QueryKey>, options?: Omit<UseQueryOptions<UseTeamRoleResponse, Error, UseTeamRoleResponse, QueryKey>, "queryFn" | "queryKey">): UseQueryResult<...>', gave the following error.
    Argument of type '() => Promise<Json>' is not assignable to parameter of type 'QueryFunction<UseTeamRoleResponse, QueryKey>'.

24     async () => {
       ~~~~~~~~~~~~~


src/utils/api/use-team-role.ts:41:24 - error TS2339: Property 'account_role' does not exist on type 'TQueryFnData'.

41     accountRole: data?.account_role,
                          ~~~~~~~~~~~~

src/utils/api/use-team-role.ts:42:27 - error TS2339: Property 'is_primary_owner' does not exist on type 'TQueryFnData'.

42     isPrimaryOwner: data?.is_primary_owner,
                             ~~~~~~~~~~~~~~~~


Found 3 errors in the same file, starting at: src/utils/api/use-team-role.ts:24
```

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -2,9 +2,6 @@
   "extends": [
     "next",
     "next/core-web-vitals",
-    "plugin:prettier/recommended"
-  ],
-  "plugins": [
     "prettier"
   ]
 }

--- a/src/components/basejump-default-content/logo.tsx
+++ b/src/components/basejump-default-content/logo.tsx
@@ -31,6 +31,7 @@ const Logo = ({ size = "lg", className }: Props) => {
           src={"/images/basejump-logo.png"}
           height={height}
           width={width}
+          alt="Basejump Logo"
         />
       </div>
       <h1

--- a/src/components/content-pages/content-layout.tsx
+++ b/src/components/content-pages/content-layout.tsx
@@ -12,7 +12,7 @@ const ContentLayout = ({ children }) => {
 
   useEffect(() => {
     toggleSidebar(false);
-  }, [router.asPath]);
+  }, [router.asPath, toggleSidebar]);
   return (
     <Drawer
       side={<ContentHeaderMobile onClose={toggleSidebar} />}

--- a/src/components/dashboard/accounts/settings/account-settings-layout.tsx
+++ b/src/components/dashboard/accounts/settings/account-settings-layout.tsx
@@ -29,7 +29,7 @@ const AccountSettingsLayout = ({ children }) => {
           router.asPath === `/dashboard/teams/${accountId}/settings/billing`,
       },
     ];
-  }, [router.pathname, accountId]);
+  }, [accountId, router.asPath, t]);
   return (
     <DashboardContent>
       <DashboardContent.Title>

--- a/src/components/dashboard/accounts/settings/invite-member.tsx
+++ b/src/components/dashboard/accounts/settings/invite-member.tsx
@@ -7,7 +7,7 @@ import { useState } from "react";
 import getInvitationUrl from "@/utils/get-invitation-url";
 import { useCopyToClipboard } from "react-use";
 import { ClipboardCopyIcon } from "@heroicons/react/outline";
-import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react";
 import { Database } from "@/types/supabase-types";
 
 type Props = {
@@ -54,6 +54,7 @@ const InviteMember = ({ accountId, onComplete }: Props) => {
   const [invitationLink, setInvitationLink] = useState<string | null>(null);
   const [state, copyToClipboard] = useCopyToClipboard();
   const { t } = useTranslation("dashboard");
+  const user = useUser();
   const supabaseClient = useSupabaseClient<Database>();
   const {
     register,
@@ -66,6 +67,7 @@ const InviteMember = ({ accountId, onComplete }: Props) => {
       .from("invitations")
       .insert({
         invitation_type: invitation.invitationType,
+        invited_by_user_id: user.id,
         account_id: accountId,
         account_role: invitation.userType,
       })

--- a/src/components/dashboard/dashboard-layout.tsx
+++ b/src/components/dashboard/dashboard-layout.tsx
@@ -25,7 +25,7 @@ const DashboardLayout = ({ children }) => {
      */
     setIsSidebarOpen(false);
     refetchDashboardOverview();
-  }, [router.asPath]);
+  }, [router.asPath, refetchDashboardOverview]);
   const currentAccount = useMemo(() => {
     if (!accountId) {
       return data?.find((a) => a.personal_account);

--- a/src/components/dashboard/sidebar/personal-account-menu.tsx
+++ b/src/components/dashboard/sidebar/personal-account-menu.tsx
@@ -34,7 +34,7 @@ const PersonalAccountMenu = ({ currentAccount }: Props) => {
       });
     }
     return internal;
-  }, [router.asPath, currentAccount]);
+  }, [router.asPath, currentAccount, t]);
   return (
     <Menu>
       {menu.map((item) => (

--- a/src/components/dashboard/sidebar/profile-button.tsx
+++ b/src/components/dashboard/sidebar/profile-button.tsx
@@ -19,7 +19,7 @@ const DashboardProfileButton = ({ className }: Props) => {
 
   const menuButtonText = useMemo(
     () => profile?.name || t("profileButton.yourAccount"),
-    [profile]
+    [profile, t]
   );
   return (
     <div className={className}>

--- a/src/components/dashboard/sidebar/team-account-menu.tsx
+++ b/src/components/dashboard/sidebar/team-account-menu.tsx
@@ -32,7 +32,7 @@ const TeamAccountMenu = ({ currentAccount }: Props) => {
       });
     }
     return items;
-  }, [router.asPath, currentAccount]);
+  }, [currentAccount, router.pathname, t]);
   return (
     <Menu>
       {menu.map((item) => (

--- a/src/components/docs/docs-layout.tsx
+++ b/src/components/docs/docs-layout.tsx
@@ -18,7 +18,7 @@ const DocsLayout = ({ navigation, children }) => {
 
   useEffect(() => {
     toggleSidebar(false);
-  }, [router.asPath]);
+  }, [router.asPath, toggleSidebar]);
 
   return (
     <Drawer

--- a/src/pages/api/og.tsx
+++ b/src/pages/api/og.tsx
@@ -5,7 +5,7 @@ export const config = {
   runtime: "experimental-edge",
 };
 
-export default function (req: NextRequest) {
+export default function SocialImage(req: NextRequest) {
   try {
     const { searchParams } = new URL(req.url);
 

--- a/src/types/supabase-types.ts
+++ b/src/types/supabase-types.ts
@@ -7,260 +7,282 @@ export type Json =
   | Json[];
 
 export interface Database {
+  graphql_public: {
+    Tables: {
+      [_ in never]: never;
+    };
+    Views: {
+      [_ in never]: never;
+    };
+    Functions: {
+      graphql: {
+        Args: {
+          operationName: string;
+          query: string;
+          variables: Json;
+          extensions: Json;
+        };
+        Returns: Json;
+      };
+    };
+    Enums: {
+      [_ in never]: never;
+    };
+  };
   public: {
     Tables: {
       account_user: {
         Row: {
-          user_id: string;
           account_id: string;
           account_role: Database["public"]["Enums"]["account_role"];
+          user_id: string;
         };
         Insert: {
-          user_id: string;
           account_id: string;
           account_role: Database["public"]["Enums"]["account_role"];
+          user_id: string;
         };
         Update: {
-          user_id?: string;
           account_id?: string;
           account_role?: Database["public"]["Enums"]["account_role"];
+          user_id?: string;
         };
       };
       accounts: {
         Row: {
+          created_at: string | null;
           id: string;
+          personal_account: boolean;
           primary_owner_user_id: string;
           team_name: string | null;
-          personal_account: boolean;
           updated_at: string | null;
-          created_at: string | null;
         };
         Insert: {
+          created_at?: string | null;
           id?: string;
+          personal_account?: boolean;
           primary_owner_user_id?: string;
           team_name?: string | null;
-          personal_account?: boolean;
           updated_at?: string | null;
-          created_at?: string | null;
         };
         Update: {
+          created_at?: string | null;
           id?: string;
+          personal_account?: boolean;
           primary_owner_user_id?: string;
           team_name?: string | null;
-          personal_account?: boolean;
           updated_at?: string | null;
-          created_at?: string | null;
         };
       };
       billing_customers: {
         Row: {
           account_id: string;
+          active: boolean | null;
           customer_id: string | null;
           email: string | null;
-          active: boolean | null;
           provider: Database["public"]["Enums"]["billing_providers"] | null;
         };
         Insert: {
           account_id: string;
+          active?: boolean | null;
           customer_id?: string | null;
           email?: string | null;
-          active?: boolean | null;
           provider?: Database["public"]["Enums"]["billing_providers"] | null;
         };
         Update: {
           account_id?: string;
+          active?: boolean | null;
           customer_id?: string | null;
           email?: string | null;
-          active?: boolean | null;
           provider?: Database["public"]["Enums"]["billing_providers"] | null;
         };
       };
       billing_prices: {
         Row: {
-          id: string;
-          billing_product_id: string | null;
           active: boolean | null;
-          description: string | null;
-          unit_amount: number | null;
+          billing_product_id: string | null;
           currency: string | null;
-          type: Database["public"]["Enums"]["pricing_type"] | null;
+          description: string | null;
+          id: string;
           interval: Database["public"]["Enums"]["pricing_plan_interval"] | null;
           interval_count: number | null;
-          trial_period_days: number | null;
           metadata: Json | null;
           provider: Database["public"]["Enums"]["billing_providers"] | null;
+          trial_period_days: number | null;
+          type: Database["public"]["Enums"]["pricing_type"] | null;
+          unit_amount: number | null;
         };
         Insert: {
-          id: string;
-          billing_product_id?: string | null;
           active?: boolean | null;
-          description?: string | null;
-          unit_amount?: number | null;
+          billing_product_id?: string | null;
           currency?: string | null;
-          type?: Database["public"]["Enums"]["pricing_type"] | null;
+          description?: string | null;
+          id: string;
           interval?:
             | Database["public"]["Enums"]["pricing_plan_interval"]
             | null;
           interval_count?: number | null;
-          trial_period_days?: number | null;
           metadata?: Json | null;
           provider?: Database["public"]["Enums"]["billing_providers"] | null;
+          trial_period_days?: number | null;
+          type?: Database["public"]["Enums"]["pricing_type"] | null;
+          unit_amount?: number | null;
         };
         Update: {
-          id?: string;
-          billing_product_id?: string | null;
           active?: boolean | null;
-          description?: string | null;
-          unit_amount?: number | null;
+          billing_product_id?: string | null;
           currency?: string | null;
-          type?: Database["public"]["Enums"]["pricing_type"] | null;
+          description?: string | null;
+          id?: string;
           interval?:
             | Database["public"]["Enums"]["pricing_plan_interval"]
             | null;
           interval_count?: number | null;
-          trial_period_days?: number | null;
           metadata?: Json | null;
           provider?: Database["public"]["Enums"]["billing_providers"] | null;
+          trial_period_days?: number | null;
+          type?: Database["public"]["Enums"]["pricing_type"] | null;
+          unit_amount?: number | null;
         };
       };
       billing_products: {
         Row: {
-          id: string;
           active: boolean | null;
-          name: string | null;
           description: string | null;
+          id: string;
           image: string | null;
           metadata: Json | null;
+          name: string | null;
           provider: Database["public"]["Enums"]["billing_providers"] | null;
         };
         Insert: {
-          id: string;
           active?: boolean | null;
-          name?: string | null;
           description?: string | null;
+          id: string;
           image?: string | null;
           metadata?: Json | null;
+          name?: string | null;
           provider?: Database["public"]["Enums"]["billing_providers"] | null;
         };
         Update: {
-          id?: string;
           active?: boolean | null;
-          name?: string | null;
           description?: string | null;
+          id?: string;
           image?: string | null;
           metadata?: Json | null;
+          name?: string | null;
           provider?: Database["public"]["Enums"]["billing_providers"] | null;
         };
       };
       billing_subscriptions: {
         Row: {
-          id: string;
           account_id: string;
-          status: Database["public"]["Enums"]["subscription_status"] | null;
+          cancel_at: string | null;
+          cancel_at_period_end: boolean | null;
+          canceled_at: string | null;
+          created: string;
+          current_period_end: string;
+          current_period_start: string;
+          ended_at: string | null;
+          id: string;
           metadata: Json | null;
           price_id: string | null;
-          quantity: number | null;
-          cancel_at_period_end: boolean | null;
-          created: string;
-          current_period_start: string;
-          current_period_end: string;
-          ended_at: string | null;
-          cancel_at: string | null;
-          canceled_at: string | null;
-          trial_start: string | null;
-          trial_end: string | null;
           provider: Database["public"]["Enums"]["billing_providers"] | null;
+          quantity: number | null;
+          status: Database["public"]["Enums"]["subscription_status"] | null;
+          trial_end: string | null;
+          trial_start: string | null;
         };
         Insert: {
-          id: string;
           account_id: string;
-          status?: Database["public"]["Enums"]["subscription_status"] | null;
+          cancel_at?: string | null;
+          cancel_at_period_end?: boolean | null;
+          canceled_at?: string | null;
+          created?: string;
+          current_period_end?: string;
+          current_period_start?: string;
+          ended_at?: string | null;
+          id: string;
           metadata?: Json | null;
           price_id?: string | null;
-          quantity?: number | null;
-          cancel_at_period_end?: boolean | null;
-          created?: string;
-          current_period_start?: string;
-          current_period_end?: string;
-          ended_at?: string | null;
-          cancel_at?: string | null;
-          canceled_at?: string | null;
-          trial_start?: string | null;
-          trial_end?: string | null;
           provider?: Database["public"]["Enums"]["billing_providers"] | null;
+          quantity?: number | null;
+          status?: Database["public"]["Enums"]["subscription_status"] | null;
+          trial_end?: string | null;
+          trial_start?: string | null;
         };
         Update: {
-          id?: string;
           account_id?: string;
-          status?: Database["public"]["Enums"]["subscription_status"] | null;
+          cancel_at?: string | null;
+          cancel_at_period_end?: boolean | null;
+          canceled_at?: string | null;
+          created?: string;
+          current_period_end?: string;
+          current_period_start?: string;
+          ended_at?: string | null;
+          id?: string;
           metadata?: Json | null;
           price_id?: string | null;
-          quantity?: number | null;
-          cancel_at_period_end?: boolean | null;
-          created?: string;
-          current_period_start?: string;
-          current_period_end?: string;
-          ended_at?: string | null;
-          cancel_at?: string | null;
-          canceled_at?: string | null;
-          trial_start?: string | null;
-          trial_end?: string | null;
           provider?: Database["public"]["Enums"]["billing_providers"] | null;
+          quantity?: number | null;
+          status?: Database["public"]["Enums"]["subscription_status"] | null;
+          trial_end?: string | null;
+          trial_start?: string | null;
         };
       };
       invitations: {
         Row: {
-          id: string;
-          account_role: Database["public"]["Enums"]["account_role"];
           account_id: string;
-          token: string;
-          invited_by_user_id: string;
+          account_role: Database["public"]["Enums"]["account_role"];
           account_team_name: string | null;
-          updated_at: string | null;
           created_at: string | null;
+          id: string;
           invitation_type: Database["public"]["Enums"]["invitation_type"];
+          invited_by_user_id: string;
+          token: string;
+          updated_at: string | null;
         };
         Insert: {
-          id?: string;
-          account_role: Database["public"]["Enums"]["account_role"];
           account_id: string;
-          token?: string;
-          invited_by_user_id: string;
+          account_role: Database["public"]["Enums"]["account_role"];
           account_team_name?: string | null;
-          updated_at?: string | null;
           created_at?: string | null;
+          id?: string;
           invitation_type: Database["public"]["Enums"]["invitation_type"];
+          invited_by_user_id: string;
+          token?: string;
+          updated_at?: string | null;
         };
         Update: {
-          id?: string;
-          account_role?: Database["public"]["Enums"]["account_role"];
           account_id?: string;
-          token?: string;
-          invited_by_user_id?: string;
+          account_role?: Database["public"]["Enums"]["account_role"];
           account_team_name?: string | null;
-          updated_at?: string | null;
           created_at?: string | null;
+          id?: string;
           invitation_type?: Database["public"]["Enums"]["invitation_type"];
+          invited_by_user_id?: string;
+          token?: string;
+          updated_at?: string | null;
         };
       };
       profiles: {
         Row: {
+          created_at: string | null;
           id: string;
           name: string | null;
           updated_at: string | null;
-          created_at: string | null;
         };
         Insert: {
+          created_at?: string | null;
           id: string;
           name?: string | null;
           updated_at?: string | null;
-          created_at?: string | null;
         };
         Update: {
+          created_at?: string | null;
           id?: string;
           name?: string | null;
           updated_at?: string | null;
-          created_at?: string | null;
         };
       };
     };
@@ -312,6 +334,135 @@ export interface Database {
         | "incomplete_expired"
         | "past_due"
         | "unpaid";
+    };
+  };
+  storage: {
+    Tables: {
+      buckets: {
+        Row: {
+          created_at: string | null;
+          id: string;
+          name: string;
+          owner: string | null;
+          public: boolean | null;
+          updated_at: string | null;
+        };
+        Insert: {
+          created_at?: string | null;
+          id: string;
+          name: string;
+          owner?: string | null;
+          public?: boolean | null;
+          updated_at?: string | null;
+        };
+        Update: {
+          created_at?: string | null;
+          id?: string;
+          name?: string;
+          owner?: string | null;
+          public?: boolean | null;
+          updated_at?: string | null;
+        };
+      };
+      migrations: {
+        Row: {
+          executed_at: string | null;
+          hash: string;
+          id: number;
+          name: string;
+        };
+        Insert: {
+          executed_at?: string | null;
+          hash: string;
+          id: number;
+          name: string;
+        };
+        Update: {
+          executed_at?: string | null;
+          hash?: string;
+          id?: number;
+          name?: string;
+        };
+      };
+      objects: {
+        Row: {
+          bucket_id: string | null;
+          created_at: string | null;
+          id: string;
+          last_accessed_at: string | null;
+          metadata: Json | null;
+          name: string | null;
+          owner: string | null;
+          path_tokens: string[] | null;
+          updated_at: string | null;
+        };
+        Insert: {
+          bucket_id?: string | null;
+          created_at?: string | null;
+          id?: string;
+          last_accessed_at?: string | null;
+          metadata?: Json | null;
+          name?: string | null;
+          owner?: string | null;
+          path_tokens?: string[] | null;
+          updated_at?: string | null;
+        };
+        Update: {
+          bucket_id?: string | null;
+          created_at?: string | null;
+          id?: string;
+          last_accessed_at?: string | null;
+          metadata?: Json | null;
+          name?: string | null;
+          owner?: string | null;
+          path_tokens?: string[] | null;
+          updated_at?: string | null;
+        };
+      };
+    };
+    Views: {
+      [_ in never]: never;
+    };
+    Functions: {
+      extension: {
+        Args: { name: string };
+        Returns: string;
+      };
+      filename: {
+        Args: { name: string };
+        Returns: string;
+      };
+      foldername: {
+        Args: { name: string };
+        Returns: string[];
+      };
+      get_size_by_bucket: {
+        Args: Record<PropertyKey, never>;
+        Returns: { size: number; bucket_id: string }[];
+      };
+      search: {
+        Args: {
+          prefix: string;
+          bucketname: string;
+          limits: number;
+          levels: number;
+          offsets: number;
+          search: string;
+          sortcolumn: string;
+          sortorder: string;
+        };
+        Returns: {
+          name: string;
+          id: string;
+          updated_at: string;
+          created_at: string;
+          last_accessed_at: string;
+          metadata: Json;
+        }[];
+      };
+    };
+    Enums: {
+      [_ in never]: never;
     };
   };
 }

--- a/src/utils/admin/stripe-billing-helpers.ts
+++ b/src/utils/admin/stripe-billing-helpers.ts
@@ -148,10 +148,10 @@ const createOrRetrieveSubscription = async ({
 
   if (!error && data) {
     return {
-      id: data.id,
-      status: data.status,
-      billing_email: data.billing_email,
-      plan_name: data.plan_name,
+      id: data["id"],
+      status: data["status"],
+      billing_email: data["billing_email"],
+      plan_name: data["plan_name"],
     };
   }
 
@@ -179,21 +179,21 @@ const createOrRetrieveSubscription = async ({
     const { data: price } = await supabaseAdmin
       .from("billing_prices")
       .select("unit_amount")
-      .eq("id", config.stripe_default_account_price_id)
+      .eq("id", config["stripe_default_account_price_id"])
       .single();
 
     if (
       !price ||
-      (price.unit_amount > 0 && !config.stripe_default_trial_period_days)
+      (price.unit_amount > 0 && !config["stripe_default_trial_period_days"])
     ) {
       throw new Error(MANUAL_SUBSCRIPTION_REQUIRED);
     }
 
     const newSubscription = await stripe.subscriptions.create({
       customer: customerId,
-      items: [{ price: config.stripe_default_account_price_id }],
+      items: [{ price: config["stripe_default_account_price_id"] }],
       expand: ["latest_invoice.payment_intent"],
-      trial_period_days: config.stripe_default_trial_period_days,
+      trial_period_days: config["stripe_default_trial_period_days"],
     });
     // now we upsert the subscription record. Upsert b/c the stripe webhook also hits this and so there could b
     await upsertSubscriptionRecord(newSubscription, accountId);
@@ -221,29 +221,29 @@ const upsertSubscriptionRecord = async (
     quantity: subscription.items.data[0].quantity,
     cancel_at_period_end: subscription.cancel_at_period_end,
     cancel_at: subscription.cancel_at
-      ? fromUnixTime(subscription.cancel_at)
+      ? fromUnixTime(subscription.cancel_at).toISOString()
       : null,
     canceled_at: subscription.canceled_at
-      ? fromUnixTime(subscription.canceled_at)
+      ? fromUnixTime(subscription.canceled_at).toISOString()
       : null,
-    current_period_start: fromUnixTime(subscription.current_period_start),
-    current_period_end: fromUnixTime(subscription.current_period_end),
-    created: fromUnixTime(subscription.created),
+    current_period_start: fromUnixTime(subscription.current_period_start).toISOString(),
+    current_period_end: fromUnixTime(subscription.current_period_end).toISOString(),
+    created: fromUnixTime(subscription.created).toISOString(),
     ended_at: subscription.ended_at
-      ? fromUnixTime(subscription.ended_at)
+      ? fromUnixTime(subscription.ended_at).toISOString()
       : null,
     trial_start: subscription.trial_start
-      ? fromUnixTime(subscription.trial_start)
+      ? fromUnixTime(subscription.trial_start).toISOString()
       : null,
     trial_end: subscription.trial_end
-      ? fromUnixTime(subscription.trial_end)
+      ? fromUnixTime(subscription.trial_end).toISOString()
       : null,
     provider: BILLING_PROVIDERS.stripe,
   };
 
   const { error } = await supabaseAdmin
     .from("billing_subscriptions")
-    .upsert([subscriptionData]);
+    .upsert(subscriptionData);
   if (error) throw error;
   console.log(
     `Inserted/updated subscription [${subscription.id}] for account [${accountId}]`

--- a/src/utils/api/use-team-role.ts
+++ b/src/utils/api/use-team-role.ts
@@ -19,7 +19,7 @@ export default function useTeamRole(
 ) {
   const user = useUser();
   const supabaseClient = useSupabaseClient<Database>();
-  const { data, isLoading } = useQuery<UseTeamRoleResponse, Error>(
+  const { data, isLoading } = useQuery(
     ["teamRole", accountId],
     async () => {
       const { data, error } = await supabaseClient

--- a/src/utils/use-auth-check.ts
+++ b/src/utils/use-auth-check.ts
@@ -17,7 +17,7 @@ const useAuthCheck = (redirectedFrom?: string) => {
     if (session && redirectedFrom) {
       replace(redirectedFrom as string);
     }
-  }, [session, redirectedFrom]);
+  }, [session, redirectedFrom, replace]);
 };
 
 export default useAuthCheck;

--- a/src/utils/use-theme-storage.ts
+++ b/src/utils/use-theme-storage.ts
@@ -12,7 +12,7 @@ export default function useThemeStorage(defaultTheme?: string) {
 
   useEffect(() => {
     setTheme(value);
-  }, [value]);
+  }, [value, setTheme]);
 
   function setInternalTheme(theme: string) {
     setValue(theme);

